napari/layers/_labels_layer/model.py

```diff
@@ -107,6 +147,130 @@ def data(self, data):
+        in all n dimensions according to specified brush size or fill.
+        """
+        return self._n_dimensional
+
+    @n_dimensional.setter
+    def n_dimensional(self, n_dimensional):
+        self._n_dimensional = n_dimensional
+        self.events.n_dimensional()
+
+        self.refresh()
+
+    @property
+    def brush_size(self):
+        """ float | list: Size of the paint brush. If a float, then if
+        `n_dimensional` is False applies just to the visible dimensions, if
+        `n_dimensional` is True applies to all dimensions. If a list, must be
+        the same length as the number of dimensions of the layer, and size
+        applies to each dimension scaled by the appropriate amount.
+        """
+        return self._brush_size
+
+    @brush_size.setter
+    def brush_size(self, brush_size):
+        self._brush_size = int(brush_size)
+        self.cursor_size = self._brush_size/self._get_rescale()
+        self.events.brush_size()
+
+        self.refresh()
+
+    @property
+    def selected_label(self):
+        """ int: Index of selected label. If `0` corresponds to the transparent
+        background. If greater than the current maximum label then if used to
+        fill or paint a region this label will be added to the new labels
+        """
+        return self._selected_label
+
+    @selected_label.setter
+    def selected_label(self, selected_label):
+        self._selected_label = selected_label
+        if selected_label == 0:
+            # If background
+            self._selected_color = None
+        else:
+            self._selected_color = self.label_color(selected_label)[0]
+        self.events.selected_label()
+
+        self.refresh()
+
+    @property
+    def mode(self):
+        """MODE: Interactive mode. The normal, default mode is PAN_ZOOM, which
+        allows for normal interactivity with the canvas.
+
+        In PICKER mode the cursor functions like a color picker, setting the
+        clicked on label to be the curent label. If the background is picked it
+        will select the background label `0`.
+
+        In PAINT mode the cursor functions like a paint brush changing any
+        pixels it brushes over to the current label. If the background label
+        `0` is selected than any pixels will be changed to background and this
+        tool functions like an eraser. The size and shape of the cursor can be
+        adjusted in the properties widget.
+
+        In FILL mode the cursor functions like a fill bucket replacing pixels
+        of the label clicked on with the current label. It can either replace
+        all pixels of that label or just those that are contiguous with the
+        clicked on pixel. If the background label `0` is selected than any
+        pixels will be changed to background and this tool functions like an
+        eraser.
+        """
+        return self._mode
+
+    @mode.setter
+    def mode(self, mode):
```

Comment:
Maybe let's start using type annotations here? If you add `mode : Mode` to the signature, tools like PyCharm will check when you try to set it to something that would trigger the ValueError.

Reply:
I'm fully open to this, but would rather do it in a followup PR where we standardize across the Shapes Layer, Markers Layer etc instead of doing it piecemeal

Reply:
Well, the whole point of them being optional is to make piecemeal adoption possible, but ok. =P